### PR TITLE
fix: zero guards cleanup + wire FFI to ApiServer pipeline

### PR DESCRIPTION
**Pull request by Subaskar-S**
## Summary

Removes all conditional compilation guards, fixes broken header guards and naming, and wires the FFI surface to the real ApiServer pipeline.

### Remove ALL `#ifdef` guards and compile definitions
- `src/api/api_server.hpp` — Remove `#ifdef GENIUS_HAS_GRPC` around `m_sgClient`
- `src/api/api_server.cpp` — Remove 3 `#ifdef` blocks (SENTENCEPIECE, GRPC init, GRPC shutdown/health)
- `src/api/CMakeLists.txt` — Remove `target_compile_definitions(GENIUS_HAS_GRPC)`
- `src/network/sg_client/sg_channel_manager.cpp` — Remove `#ifdef GENIUS_HAS_GRPC` wrapper
- `src/network/CMakeLists.txt` — Uncomment `super_genius_client.cpp`, remove `GENIUS_HAS_LIBP2P`
- `src/core/tokenizer/sentence_piece_tokenizer.cpp` — Remove duplicate `#ifdef` guards
- `src/core/CMakeLists.txt` — Remove 4 `target_compile_definitions` (MNN, SENTENCEPIECE, VULKAN, SGPROCESSING)

### Repair header guards and naming
- Fix `GENIUSAPISERVER` → `API_SERVER` in `api_server.hpp`
- Fix broken `\1_H` comment → `GNUS_NEO_SWARM_GENIUS_ELM_CHAT_C_H`
- Rename `genius_test` → `neoswarm_test` (macro + 11 call sites in `test/CMakeLists.txt`)

### Remove hardcoded vocab size
- Remove anonymous namespace `kDefaultVocabSize = 32000U` from `mnn_inference_engine.cpp`
- Stub path now returns `Error::TokenizerFailed` instead of generating random logits

### Wire FFI to real ApiServer pipeline
- `GeniusElmInit` creates and initializes an ApiServer with model/knowledge paths
- `GeniusElmChatCompletionsCreate` parses JSON via nlohmann::json, dispatches to real pipeline
- `GeniusElmGetStatus` returns live status (running, connected, fallback state)
- Falls back to stub responses when server is not initialized
- Thread-safe via global mutex
- Links `Genius-MOS-ELM-FFI` against `neoswarm_api`

### CODE REVIEW

File: src/api/api_server.hpp [hunk 1]
```diff
@@ -4,8 +4,8 @@
  * @date       2026-05-08
  */
 
-#ifndef NEOSWARM_API_GENIUSAPISERVER_HPP
-#define NEOSWARM_API_GENIUSAPISERVER_HPP
+#ifndef NEOSWARM_API_API_SERVER_HPP
+#define NEOSWARM_API_API_SERVER_HPP
```
Comment (henriqueaklein) [inline src/api/api_server.hpp]: Weird naming. Duplicated API

File: src/genius_elm_chat_completions.cpp [hunk 2]
```diff
@@ -32,40 +58,150 @@ namespace
   }
 })";
 
-    constexpr const char* kStatusJson = R"({
+    const char* kStatusJsonStub = R"({
```
Comment (henriqueaklein) [inline src/genius_elm_chat_completions.cpp]: better to have this as a a const std::string_view kStatusJsonStub

File: src/genius_elm_chat_completions.cpp [hunk 3]
```diff
@@ -75,6 +211,7 @@ extern "C"
 
     GENIUS_ELM_CHAT_C_API char* GeniusElmGetStatus( void ) GENIUS_ELM_CHAT_C_NOEXCEPT
     {
-        return AllocCopy( kStatusJson );
+        std::lock_guard<std::mutex> lock( g_mutex );
```
Comment (henriqueaklein) [inline src/genius_elm_chat_completions.cpp]: g_mutex is a mutex used to access g_server? if so, it should be used when g_server is accessed. So this mutex should be moved to BuildStatusJson

File: src/genius_elm_chat_completions.h [hunk 4]
```diff
@@ -77,4 +85,4 @@ extern "C"
 }
 #endif
```
Comment (henriqueaklein) [inline src/genius_elm_chat_completions.h:86]: GeniusElmStringFree should be here, otherwise the user can't free the responses it got from GeniusElmChatCompletionsCreate and GeniusElmGetStatus

**Review by henriqueaklein — CHANGES_REQUESTED** [review]

**Comment by Subaskar-S** [comment]
Thanks for the review and detailed feedback. I will check these points and update/fix them accordingly.

File: src/api/api_server.hpp (hunk 1 above)
Comment (Subaskar-S) [reply src/api/api_server.hpp]: Fixed, thank you. Renamed the header guard to NEOSWARM_API_SERVER_HPP

File: src/genius_elm_chat_completions.cpp (hunk 2 above)
Comment (Subaskar-S) [reply src/genius_elm_chat_completions.cpp]: Done. Changed both kStubChatJson and kStatusJsonStub to constexpr std::string_view.

File: src/genius_elm_chat_completions.cpp (hunk 3 above)
Comment (Subaskar-S) [reply src/genius_elm_chat_completions.cpp]: Fixed. Moved the std::lock_guard into BuildStatusJson() where g_server is accessed and removed the redundant lock from GeniusElmGetStatus(). Other callers retain their locks because they access g_server directly.

**Review by henriqueaklein — APPROVED** [review]
